AndroidHomeApplication/app/src/main/java/com/example/androidhomeapplication/App.kt

```diff
@@ -1,19 +1,22 @@
 package com.example.androidhomeapplication
 
 import android.app.Application
-import com.android.academy.fundamentals.homework.data.JsonMovieRepository
-import com.android.academy.fundamentals.homework.data.MovieRepository
-import com.android.academy.fundamentals.homework.data.MovieRepositoryProvider
+import com.example.androidhomeapplication.data.remote.ApiClient
+import com.example.androidhomeapplication.data.remote.RetrofitBuilder
+import com.example.androidhomeapplication.data.repository.RepositoryProvider
+import com.example.androidhomeapplication.data.repository.MoviesRepository
 import com.example.androidhomeapplication.navigation.NavigatorHolderProvider
 import com.example.androidhomeapplication.navigation.RouterProvider
 import com.github.terrakok.cicerone.Cicerone
 import com.github.terrakok.cicerone.NavigatorHolder
 import com.github.terrakok.cicerone.Router
 
-class App : Application(), RouterProvider, NavigatorHolderProvider, MovieRepositoryProvider {
+class App : Application(), RouterProvider, NavigatorHolderProvider, RepositoryProvider {
     private val cicerone = Cicerone.create()
 
     override val router: Router get() = cicerone.router
     override val navigatorHolder: NavigatorHolder get() = cicerone.getNavigatorHolder()
-    override val movieRepository: MovieRepository by lazy { JsonMovieRepository(this) }
+
+    private val apiClient: ApiClient = ApiClient(RetrofitBuilder.buildRetrofit())
+    override val movieRepository: MoviesRepository = MoviesRepository(apiClient)
```

Comment:
override val movieRepository: MoviesRepository = MoviesRepository(ApiClient(RetrofitBuilder.buildRetrofit()))

as private val apiClient: ApiClient not used, but reference is hold